### PR TITLE
[ENH] replace `sklearn` imports of metrics by `sktime` classes

### PR DESCRIPTION
**Pull request by fkiraly**
This PR replaces `sklearn` imports of metrics by `sktime` classes - reducing the dependency footprint of the metrics module on `sklearn` significantly, but not entirely.

This should be a pure refactor, logic-wise, without impact on the public APIs.

The motivation is the bug in https://github.com/sktime/sktime/issues/8457 resulting from instability of `sklearn` public APIs, of metrics.

### CODE REVIEW

**Comment by fkiraly** [comment]
FYI @andretheronsa

**Comment by fkiraly** [comment]
well, this is the structural approach at least - imo the idea should work, but someone needs to debug these errors... I might do this at some point.

**Comment by yarnabrina** [comment]
Just found this PR, may I ask what is the pros/cons of removing sklearn metric footprint? Is there any issue in current metrics usage wise?

**Comment by fkiraly** [comment]
> Just found this PR, may I ask what is the pros/cons of removing sklearn metric footprint?
> Is there any issue in current metrics usage wise?

Lack of stability in the extension contract is the main problem. In one instance (1.7.0) this has even rendered the metric functions completely inoperable.

See here for the wider problem related to extension contracts: https://github.com/scikit-learn/scikit-learn/issues/31728, and for the concrete issue that triggered this, here: https://github.com/sktime/sktime/issues/8457

High-level, many extension contracts - even if they are official (the exact status is not clear) - rely on private functions from the repository, such as `_validate_data` (without this, e.g., an `sklearn` estimator cannot be written in compliant fashion!). Now since they are private, the `sklearn` devs consider it to be fair game to break in any minor release, which effectively makes the extension contracts unsafe.

Since the recommended way is using the classes anyway - e.g., due to applicability to hierarchical data - it is perhaps best to remove the `sklearn` footprint by pointing to the classes as a the primary object here.
